core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,855 @@
+    std::string const dbKey = _prefix + "/DBS/" + dbName;
+    _storePacked(dbKey, dbMap);
+    _kvI->set(dbKey, "READY");
+}
+
+void
+CssAccess::dropDb(std::string const& dbName) {
+    LOGF(_log, LOG_LVL_DEBUG, "dropDb(%1%)" % dbName);
+
+    std::string key = _prefix + "/DBS/" + dbName;
+
+    // key is supposed to exist and key.json is optional
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete packed key: %s.json" % key);
+        _kvI->deleteKey(key + ".json");
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: packed key is not there");
+    }
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete regular key: %s" % key);
+        _kvI->deleteKey(key);
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: regular key is not there");
+        throw NoSuchDb(dbName);
+    }
+}
+
+std::vector<std::string>
+CssAccess::getTableNames(std::string const& dbName, bool readyOnly) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getTableNames(%1%)" % dbName);
+
+    std::string key = _prefix + "/DBS/" + dbName + "/TABLES";
+    std::vector<std::string> names;
+    try {
+        names = _kvI->getChildren(key);
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "getTableNames: key is not found: %s" % key);
+        throw NoSuchDb(dbName);
+    }
+
+    // remove names from packed keys
+    auto it = std::remove_if(names.begin(), names.end(),
+                             [](const std::string& name) {
+                                return name.compare(name.size()-5, name.size(), ".json") == 0; });
+    names.erase(it, names.end());
+
+    if (readyOnly and not names.empty()) {
+        // filter out names with status other than READY
+        auto const tableStatuses = _getSubkeys(key, names);
+        names.clear();
+        for (auto& kv: tableStatuses) {
+            if (kv.second == "READY") {
+                names.push_back(kv.first);
+            }
+        }
+    }
+    return names;
+}
+
+bool
+CssAccess::containsTable(std::string const& dbName, std::string const& tableName, bool readyOnly) const {
+    LOGF(_log, LOG_LVL_DEBUG, "containsTable(%1%, %2%)" % dbName % tableName);
+
+    std::string const key = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
+    // If key is not there pretend that its value is not "READY"
+    std::string const val = _kvI->get(key, "DOES_NOT_EXIST");
+    if (val == "DOES_NOT_EXIST") {
+        // table key is not there at all, throw if database name is not good
+        _assertDbExists(dbName);
+        LOGF(_log, LOG_LVL_DEBUG, "containsTable: key is not there: %1%" % key);
+        return false;
+    }
+    LOGF(_log, LOG_LVL_DEBUG, "containsTable: key value: %1%" % val);
+    // if key value is not "READY" it likely means table is in the process
+    // of being deleted, which is the same as if it does not exist
+    if (readyOnly) return val == "READY";
```

Comment:
Our coding guidelines has example of one-line `if`without braces, so it should be ok.